tests/07-single-hop-udp/app/main.c

```diff
@@ -0,0 +1,178 @@
+ */
+
+/**
+ * @ingroup     examples
+ * @{
+ *
+ * @file
+ * @brief       Example application for demonstrating the RIOT network stack
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include <stdio.h>
+
+#include "shell.h"
+#include "msg.h"
+
+#include "net/gnrc.h"
+#include "net/gnrc/ipv6.h"
+#include "net/gnrc/udp.h"
+
+#define PKTSIZE         (1000U)
+#define ITER            (1000)
+#define DELAY           (1000000U)
+#define PORT            (12345U)
+
+#define PRIO            (THREAD_PRIORITY_MAIN - 1)
+
+static uint8_t load[PKTSIZE];
+static msg_t msg_queue[8];
+static char stack[THREAD_STACKSIZE_DEFAULT];
+
+static volatile int cnt = 0;
+
+static gnrc_netreg_entry_t server = { NULL, GNRC_NETREG_DEMUX_CTX_ALL, KERNEL_PID_UNDEF };
+
+void udp_send(ipv6_addr_t *addr, uint16_t p, uint8_t *data, size_t len)
+{
+    uint8_t port[2];
+
+    /* parse port */
+    port[0] = (uint8_t)p;
+    port[1] = p >> 8;
+
+    gnrc_pktsnip_t *payload, *udp, *ip;
+    /* allocate payload */
+    payload = gnrc_pktbuf_add(NULL, data, len, GNRC_NETTYPE_UNDEF);
+    if (payload == NULL) {
+        puts("UDP Error: unable to copy data to packet buffer");
+        return;
+    }
+    /* allocate UDP header, set source port := destination port */
+    udp = gnrc_udp_hdr_build(payload, port, 2, port, 2);
+    if (udp == NULL) {
+        puts("UDP Error: unable to allocate UDP header");
+        gnrc_pktbuf_release(payload);
+        return;
+    }
+    /* allocate IPv6 header */
+    ip = gnrc_ipv6_hdr_build(udp, NULL, 0, addr->u8, sizeof(ipv6_addr_t));
+    if (ip == NULL) {
+        puts("UDP Error: unable to allocate IPv6 header");
+        gnrc_pktbuf_release(udp);
+        return;
+    }
+    /* send packet */
+    if (!gnrc_netapi_dispatch_send(GNRC_NETTYPE_UDP, GNRC_NETREG_DEMUX_CTX_ALL, ip)) {
+        puts("UDP Error: unable to locate UDP thread");
+        gnrc_pktbuf_release(ip);
+        return;
+    }
+}
+
+static void *svr_thread(void *arg)
+{
+    (void)arg;
+    msg_t msg;
+    gnrc_pktsnip_t *snip;
+
+    /* setup the message queue */
+    msg_init_queue(msg_queue, 8);
+
+    while (1) {
+        msg_receive(&msg);
+
+        switch (msg.type) {
+            case GNRC_NETAPI_MSG_TYPE_RCV:
+                snip = (gnrc_pktsnip_t *)msg.content.ptr;
+                    if (snip->size != PKTSIZE) {
+                        puts("SVR: UDP packet has invalid size");
+                    }
+                    else {
+                        printf("got pkt #%i\n", cnt++);
+                    }
+                    gnrc_pktbuf_release(snip);
+                break;
+            default:
+                puts("SVR: got invalid message");
+                break;
+        }
+    }
+
+    /* never reached */
+    return NULL;
+}
+
+static int cmd_run(int argc, char **argv)
+{
+    uint32_t last = xtimer_now();
+    ipv6_addr_t addr;
+
+    if (argc != 2) {
+        printf("usage: %s <target IP address>\n", argv[0]);
```

Comment:
I would prefer, to be able to still set the number, the size and the interval externally. You can provide a pexpect script that pre-set this.